src/main/kotlin/com/dansplugins/factionsystem/gate/MfGateService.kt

```diff
@@ -56,9 +56,21 @@ class MfGateService(
     fun getGatesByStatus(status: MfGateStatus) = gatesById.values.filter { it.status == status }
 
     fun save(gate: MfGate) = resultFrom {
-        val result = gateRepo.upsert(gate)
-        gatesById[result.id] = result
-        return@resultFrom result
+        try {
+            val result = gateRepo.upsert(gate)
+            gatesById[result.id] = result
+            return@resultFrom result
+        } catch (e: OptimisticLockingFailureException) {
+            plugin.logger.warning("Optimistic locking failure for gate with ID: ${gate.id}. Retrying...")
+            val reloadedGate = gateRepo.getGate(gate.id) ?: throw IllegalStateException("Gate not found: ${gate.id}")
+            val mergedGate = reloadedGate.copy(
+                status = gate.status,
```

Comment:
There's a lot more than just status that could be updated - how do we know this is all that needs updating?